src/hooks/useAddToCart.ts

```diff
@@ -0,0 +1,24 @@
+// import { useEffect } from 'react';
+import { useDispatch /* useSelector */ } from 'react-redux';
+
+// import { RootState } from '@/app/store';
+
+import { addToCart, toggleClickedBuy } from '../features/сart/CartSlice';
+import { Product } from '../types/Phone';
+
+export const useAddCartButton = (product: Product) => {
```

Comment:
It's better to make one hook for this like useCart, there's no point in spreading it out over different ones